Core/Object Arts/Dolphin/Base/String.cls

```diff
@@ -878,11 +878,11 @@ printOn: aStream
 				ifFalse: [(aStream nextPut: ch) == $' ifTrue: [aStream nextPut: $']]].
 	aStream nextPut: $'!
 
-refersToLiteral: anObject 
+refersToLiteral: anObject
 	"Private - Answer whether the receiver is a reference to the literal argument.
 	This assumes that the receiver is in the role of a literal."
 
-	^self = anObject!
+	^anObject isSymbol not and: [self = anObject].!
```

Comment:
I'm not keen on the type test, but I suppose it can't be avoided without introducing a double-dispatch, which would be overkill for this. Oh well...

Reply:
Right, yeah, it bothered me too, but I couldn't see a better way.